File: Medium/Java/377. Combination Sum IV.java
class Solution {
    public int combinationSum4(int[] nums, int target) {
        int[] cases = new int[target + 1];
        cases[0] = 1;

        for(int i = 1; i <= target; i++){
            for(int n : nums){
                if(n <= i)
                    cases[i] += cases[i - n];
            }
        }

        return cases[target];
    }
}

File: Medium/Python/377. Combination Sum IV.py
class Solution(object):
    def combinationSum4(self, nums, target):
        cases = [0] * (target + 1)
        cases[0] = 1

        for i in range(1, target + 1):
            for n in nums:
                if n <= i:
                    cases[i] += cases[i - n]
        
        return cases[target]
